src/Jackalope/Node.php

```diff
@@ -773,9 +785,10 @@ public function getPropertiesValues($nameFilter = null, $dereference = true)
         $this->checkState();
 
         // OPTIMIZE: do not create properties in constructor, go over array here
-        $names = self::filterNames($nameFilter, array_keys($this->properties));
+        $names = self::filterNames($nameFilter, array_keys($this->propertyValues));
         $result = array();
         foreach ($names as $name) {
+            $this->loadProperty($name);
```

Comment:
i think the idea of getPropertiesValues was that you could get the values without needing to instantiate the properties. do we still need this call to do cleanup?